src/codeql.ts

```diff
@@ -55,6 +55,8 @@ async function runQuery(
   console.log(
     `This database was created using CodeQL CLI version ${dbMetadata.creationMetadata?.cliVersion}`
   );
+  const databaseSHAFile = path.join("results", "sha.txt");
+  fs.writeFileSync(databaseSHAFile, dbMetadata.creationMetadata?.sha || "");
```

Comment:
The failure case is the only thing that I'm unsure about here.

How would you expect a `SuccessIndexItem` to look if we don't know the sha? I was thinking that perhaps it should be declared as optional in that interface and then to not to include it if either `sha.txt` exists and contains the empty string or doesn't exist at all (as you prefer).